gooddata-sdk/gooddata_sdk/catalog/workspace/service.py

```diff
@@ -806,6 +810,149 @@ def set_translated_texts(
+
+        Args:
+            workspace_id (str): The ID of the workspace.
+            target_language (str): The target language for the metadata localization.
+            translator_func (Optional[Callable]): A function to translate the source text.
+            set_locale (bool): Flag to indicate if the locale settings should be updated in the workspace.
+
+        Returns:
+            None
+        """
+        ans = self._actions_api.retrieve_translations(
+            workspace_id=workspace_id,
+            locale_request=LocaleRequest(locale=target_language),
+            _preload_content=False,
+            async_req=False,
+        )
+
+        encoded_xml = self._add_target_tags(ans.data.decode(), translator_func)
+
+        self.set_metadata_localisation(workspace_id=workspace_id, encoded_xml=encoded_xml)
+        if set_locale:
+            metadata_locale = "METADATA_LOCALE"
+            locale = "LOCALE"
+
+            self.create_or_update_workspace_setting(
+                workspace_id,
+                CatalogWorkspaceSetting(
+                    id=metadata_locale, setting_type=metadata_locale, content={"value": target_language}
+                ),
+            )
+            self.create_or_update_workspace_setting(
+                workspace_id,
+                CatalogWorkspaceSetting(id=locale, setting_type=locale, content={"value": target_language}),
+            )
+
+    def save_metadata_locale_to_disk(self, workspace_id: str, target_language: str, file_path: Path) -> None:
+        """Save the metadata localization for a workspace to a file.
+
+        Args:
+            workspace_id (str): The ID of the workspace.
+            target_language (str): The target language for the metadata localization.
+            file_path (Path): The path to the file where the XLIFF content will be saved.
+
+        Returns:
+            None
+        """
+        xliff_content = self._actions_api.retrieve_translations(
+            workspace_id=workspace_id,
+            locale_request=LocaleRequest(locale=target_language),
+            _preload_content=False,
+        ).data.decode()
+
+        ns = get_namespace_from_xml(xliff_content)
+
+        ET.register_namespace("", ns["ns"])
+        tree = ET.ElementTree(ET.fromstring(xliff_content))
+
+        tree.write(file_path, "utf-8")
+
+    def load_metadata_locale_from_disk(self, workspace_id: str, file_path: Path) -> None:
+        """Load and set the metadata localization for a workspace from a file.
+
+        Args:
+            workspace_id (str): The ID of the workspace to which the metadata localization applies.
+            file_path (Path): The path to the file containing the encoded XML metadata.
+
+        Returns:
+            None
+        """
+        with open(file_path, "rb") as f:
+            encoded_xml = f.read()
+            self.set_metadata_localisation(workspace_id=workspace_id, encoded_xml=encoded_xml)
```

Comment:
The method is called load, but it does return anything. I propose to rename it to `set_metadata_locale_from_disk`.